safe/gui/tools/batch/batch_dialog.py

```diff
@@ -271,99 +280,252 @@ def run_script(self, filename):
         else:
             function()
 
-    def run_scenario(self, items):
-        """Run a simple scenario.
-
-        :param items: A dictionary containing the scenario configuration
-            as table items.
-        :type items: dict
-
-        :return: A tuple with the status of the IF and an error message if
-            needed.
-            The status is ANALYSIS_SUCCESS if everything was fine.
-            The status is ANALYSIS_FAILED_BAD_INPUT if the client should fix
-                something.
-            The status is ANALYSIS_FAILED_BAD_CODE if something went wrong
-                from the code.
-        :rtype: (int, m.Message)
+    def reset_status(self):
+        """Set all scenarios' status to empty in the table
         """
-        # LOGGER.info('Run simple task' + str(items))
-        scenario_directory = str(self.source_directory.text())
+        for row in range(self.table.rowCount()):
+            status_item = self.table.item(row, 1)
+            status_item.setText(self.tr(''))
 
-        paths = []
-        if 'aggregation' in items:
-            paths.append(items['aggregation'])
-        if 'exposure' in items:
-            paths.append(items['exposure'])
-            # add access to exposure layer source so we can access it later
-            self.exposure_source = os.path.normpath(
-                os.path.join(scenario_directory, items['exposure']))
-        if 'hazard' in items:
-            paths.append(items['hazard'])
+    def prepare_task(self, items):
+        """Prepare scenario so that it can be set directly to impact
+        function variable
 
-        # always run in new project
-        if self.start_in_new_project:
-            self.iface.newProject()
+        :param items: dictionary containing settings for impacr function
```

Comment:
* `impact`
* It's also a good habit to start with capital letter and finish with a full stop for docstrings.